azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -966,16 +994,24 @@ private void pollExecution() {
           }
         }
       } else if (FlowRunnerManager.this.active) {
-        try {
-          // Todo jamiesjc: check executor capacity before polling from DB
-          final int execId = FlowRunnerManager.this.executorLoader
-              .selectAndUpdateExecution(this.executorId);
-          if (execId != -1) {
-            FlowRunnerManager.logger.info("Submitting flow " + execId);
-            submitFlow(execId);
+        FlowRunnerManager.this.preparingFlowCount.getAndIncrement();
+        if (FlowRunnerManager.this.active) {
+          try {
+            // Todo jamiesjc: check executor capacity before polling from DB
+            final int execId = FlowRunnerManager.this.executorLoader
+                .selectAndUpdateExecution(this.executorId);
+            if (execId != -1) {
+              FlowRunnerManager.logger.info("Submitting flow " + execId);
+              Thread.sleep(1000 * 30);
+              submitFlow(execId);
+            }
+          } catch (final Exception e) {
+            FlowRunnerManager.logger.error("Failed to submit flow ", e);
+          } finally {
+            FlowRunnerManager.this.preparingFlowCount.getAndDecrement();
           }
-        } catch (final Exception e) {
-          FlowRunnerManager.logger.error("Failed to submit flow ", e);
+        } else {
+          FlowRunnerManager.this.preparingFlowCount.getAndDecrement();
         }
```

Comment:
Could a unit test be added.

Reply:
will add todo given poor testability of this class.

Reply:
https://github.com/azkaban/azkaban/issues/2134

Reply:
Could you please also file a JIRA ticket for tracking?

Reply:
created internally.